crates/jstzd/tests/octez_node_test.rs

```diff
@@ -0,0 +1,77 @@
+use jstzd::task::{octez_node, Task};
+
+async fn retry<'a>(
+    retries: u16,
+    interval_ms: u64,
+    f: &(dyn Fn() -> std::pin::Pin<
+        Box<dyn std::future::Future<Output = anyhow::Result<bool>> + Send + 'a>,
+    > + Sync),
+) -> bool {
```

Comment:
```suggestion
async fn retry<'a, F>(retries: u16, interval_ms: u64, f: impl Fn() -> F) -> bool
where
    F: Future<Output = anyhow::Result<bool>> + Send + 'a,
{
```

Reply:
Prefer impl Trait over dyn Trait for compile time polymorphism.

Reply:
Updated